### PR TITLE
Created single entry point for using, building and testing puzzle tools

### PR DESCRIPTION
**Pull request by s-zhang**

### CODE REVIEW

File: puzzle-utils/puzzle-utils.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+from words.anagram import anagram, anagram_multi
+from words.search import search
+
+# Recommend to use this interactively by running
+# >python -i puzzle-utils.py
+
+def main():
+	"""
+		Examples:
+		words.utils.dictionary.CURRENT_DICTIONARY = Dictionary("google-10000-english-usa")
```
[the remark this answers is not in the document]
Comment (s-zhang) [reply puzzle-utils/puzzle-utils.py]: Make sense. Will address

File: readme.md [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
-## Tools to Create
+## puzzle-utils
+
+## Features/To-Dos
 (Sign up for whatever you decide to work on here so we don't duplicate work)
 
 Basically what I want is all the random online tools in one easy lightweight place or in script version.
 
-- Word Ladder tools
+- Word Ladder tools (TODO)
```
Comment (naarcini) [inline readme.md]: Wasn't the plan to open issues on these?

File: readme.md [hunk 3]
```diff
@@ -20,8 +22,53 @@ Basically what I want is all the random online tools in one easy lightweight pla
...
+
+Refer to [puzzle-utils/README.md](puzzle-utils/README.md) for details.
+
+## Authors
+
+See [contributors](https://github.com/s-zhang/puzzlehunt-tools/contributors).
+
+## License
+
+TBD
```
Comment (naarcini) [inline readme.md]: May I suggest [GPL-3.0](https://opensource.org/licenses/GPL-3.0)

File: puzzle-utils/words/anagram.py [hunk 4]
```diff
@@ -1,5 +1,5 @@
-from utils.dictionary import get_current_dictionary
-from utils.word import get_character_histogram, histogram_to_string
+from .utils.dictionary import get_current_dictionary
```
Comment (naarcini) [inline puzzle-utils/words/anagram.py]: Is the leading `.` really necessary?

File: puzzle-utils/words/anagram.py [hunk 5]
```diff
@@ -1,5 +1,5 @@
...
+from .utils.word import get_character_histogram, histogram_to_string
 
 class AnagramPattern:
 	num_of_letters = 26
```
Comment (naarcini) [inline puzzle-utils/words/anagram.py]: nit: Prefer either `number_of_letters` or `num_letters`. If we're shortening variable names, may as well remove prepositions

File: puzzle-utils/puzzle-utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (naarcini) [reply puzzle-utils/puzzle-utils.py]: Better yet, the dictionary type should be an enum or something similar so I don't have to type `Dictionary("Peter Norvig in Collins")`

File: readme.md (hunk 2 above)
Comment (s-zhang) [reply readme.md]: I realized it'd be nice for people to see what are the features that are and will be in this toolset. We can still create issues to track progress, thoughts on these items.

File: puzzle-utils/words/anagram.py (hunk 4 above)
Comment (s-zhang) [reply puzzle-utils/words/anagram.py]: Yeah. Python oddity with modules here. O.w. it won't work.

File: readme.md (hunk 3 above)
Comment (s-zhang) [reply readme.md]: This should be a separate discussion. See issue https://github.com/s-zhang/puzzlehunt-tools/issues/8

File: puzzle-utils/puzzle-utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (s-zhang) [reply puzzle-utils/puzzle-utils.py]: Good point. I added constants for the dictionaries and added tab completion so you don't have to type the whole thing.

**Pull request merged**